### PR TITLE
error propagation on HTTP and HTTPS frontend add and remove

### PR DESCRIPTION
**Pull request by Keksoj**
In the spirit of #825 , we prefer one-line context-full errors logs instead of several lines apart from one another.

This PR adds error propagation to HTTP and HTTPS frontend adding & removing logic.

### CODE REVIEW

**Comment by Keksoj** [comment]
merge #861 first, and then rebase this

File: lib/src/http.rs [hunk 1]
```diff
@@ -948,22 +948,28 @@ impl Session {
...
+        let route = match cluster_id_res {
+            Ok(route) => route,
+            Err(e) => {
+                let no_host_error = format!("Host not found: {}: {:#}", host, e);
                 self.set_answer(DefaultAnswerStatus::Answer404, None);
                 bail!(no_host_error);
             }
         };
 
+        let cluster_id = match route {
```
Comment (FlorentinDUBOIS) [inline lib/src/http.rs:965]: To me, this should be merge with the above match like:

```rust
match cluster_id_res {
Ok(Router::ClusterId(cluster_id)) => ...
Ok(Route::Deny) => ..
Err(err) => ..
}
```

File: lib/src/https.rs [hunk 2]
```diff
@@ -1227,25 +1227,31 @@ impl Session {
...
-            None => {
-                let no_host_error = format!("Host not found: {}", host);
+        let route = match cluster_id_res {
+            Ok(route) => route,
+            Err(e) => {
+                let no_host_error = format!("Host not found: {}: {:#}", host, e);
                 self.set_answer(DefaultAnswerStatus::Answer404, None);
                 bail!(no_host_error);
             }
+        };
```
Comment (FlorentinDUBOIS) [inline lib/src/https.rs:1247]: Same as above, match could be nested ;)

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]

File: lib/src/http.rs [hunk 3]
```diff
@@ -948,22 +948,28 @@ impl Session {
             .borrow()
             .listeners
             .get(&self.listener_token)
+            .with_context(|| "No listener found for this request")? // very unlikely
```
[the remark this answers is not in the document]
Comment (Keksoj) [reply lib/src/http.rs]: removed

File: lib/src/http.rs (hunk 1 above)
Comment (Keksoj) [reply lib/src/http.rs]: done

File: lib/src/https.rs (hunk 2 above)
Comment (Keksoj) [reply lib/src/https.rs]: done

**Comment by Keksoj** [comment]
At last, it is ready for review and merge @FlorentinDUBOIS ;-)

**Review by FlorentinDUBOIS — APPROVED** [review]

**Pull request merged**